cupy/lib/_routines_poly.py

```diff
@@ -114,6 +115,121 @@ def _polypow(x, n):
     return cupy.convolve(x, _polypow(cupy.convolve(x, x), (n - 1) // 2))
 
 
+def _polyfit_typecast(x):
+    if x.dtype == numpy.float16:
+        raise NotImplementedError('float16 is currently not supported')
```

Comment:
As temporary support, how about casting the float16 input to float32 input, and casting the result to correct dtype?

Reply:
float16 is not supported in `numpy.polyfit` because of `numpy.linalg.lstsq`. `TypeError` should have been raised instead of `NotImplementedError`. I fixed it in the last commit and added a test.

Reply:
Could you check it with `accpet_error`? (https://github.com/cupy/cupy/pull/3747#discussion_r476592458)

Reply:
It is no possible to use accept_error for float16 in `TestPolyfitDiffTypes` tests.

Reply:
Why?

Reply:
I committed some fixes. It will be difficult to support `float16` x and `bool` y combination. I raised `NotImplementedError` for this case. So either `no_float16` or `no_bool` shall be used in `DiffTypes` tests. PTAL